src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -76,7 +78,7 @@ public Object processRequest(Request request, Response response, String userId)
                     if (StringUtils.isNotBlank(requestBody)) {
                         JSONObject jsonObject = new JSONObject(requestBody);
                         String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
-                        Integer userIdRequest = Integer.parseInt(userIdR);
+                        Long userIdRequest = Long.parseLong(userIdR);
```

Comment:
The comment is not related to this specific line but as I see this class is written exactly like `FilterRoute` was, maybe it is good time to refactor it, before it will become as ugliest as `FilterRoute` was?

Reply:
I agree with the refactoring in general. I will put that on my list, not sure if we have to do it in this ticket or not since I think its good to have this tested and merged. 
@SimoneMaiwald what do you think?

Reply:
most of the time postponed refactoring ends up with the hope that some day someone will refactor it and after a while we have doubled size messed up code than something related to refactoring.

Reply:
yes, I think we should refactor it now, it is a good point now that we have bundle pdf and this change, I think it is time to do it and then also catch the not able to merge error better. with a nicer message in slack (maybe user and pt id and what pdf)

Reply:
@pegahtah separation looks good, but route itself still looks more or less complicated, because of lots of if/if chaining, I would recommend to use Factory Object design pattern, factory class which would be responsible to generate different pdf classes, it would help to release route class from `if`s, also try to use guard condition approach instead of putting huge logic in if blocks, better approach of guard condition is `Optional` if it is possible to use.